tuning/sft_trainer.py

```diff
@@ -263,7 +233,7 @@ def train(
         peft_config=peft_config,
     )
 
-    if run_distributed and peft_config is not None:
+    if trainer.is_fsdp_enabled and peft_config is not None:
```

Comment:
does this flag `trainer.is_fsdp_enabled` basically do the same thing where its enabled only in a multi GPU env that has the desired HW and disabled for single gPU env?

Reply:
this is the API correct way to check if FSDP is enabled. it is [initialized like this](https://github.com/huggingface/transformers/blob/45c065109074d60c587d3e562f16531d02a422f6/src/transformers/trainer.py#L4101) from the `trainer.accelerator.state`, which is the official way HF handles FSDP configs.